src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -1322,12 +1317,31 @@ export function clinicalDataCountComparator(a: ClinicalDataCount, b: ClinicalDat
     }
 }
 
-// Descent sort priority then ascent sort by display name
-export function chartMetaComparator(a: ChartMeta, b: ChartMeta): number
-{
+type ClinicalAttributeSorting = {
+    priority: number,
+    displayName: string
+}
+
+export function clinicalAttributeSortingComparator(a: ClinicalAttributeSorting, b: ClinicalAttributeSorting): number {
     return b.priority - a.priority || a.displayName.localeCompare(b.displayName);
 }
 
+export function clinicalAttributeComparator(a: ClinicalAttribute, b: ClinicalAttribute): number {
+    return clinicalAttributeSortingComparator({
+        priority: Number(a.priority),
+        displayName: a.displayName
+    }, {
+        priority: Number(b.priority),
+        displayName: b.displayName
+    });
+}
+
+// Descent sort priority then ascent sort by display name
+export function chartMetaComparator(a: ChartMeta, b: ChartMeta): number {
+    return clinicalAttributeSortingComparator(_.pick(a, ['priority', 'displayName']), _.pick(b, ['priority', 'displayName']))
```

Comment:
not typesafe b/c props passed as string.   porbably just just do   ` {  priority:a.priority, displayName:a.displayName  } `
but actually, why not just pass the whole object?  why do you need to get rid of the other things?

Reply:
The function is used in two places but the models are different. They just have common properties priority and displayName which are only required component.